modules/efk-logging-deploy-subscription.adoc

```diff
@@ -124,31 +124,32 @@ spec:
   logStore:
     type: "elasticsearch"  <3>
     elasticsearch:
-      nodeCount: 3
+      nodeCount: 3 <4>
       storage:
         storageClassName: gp2
         size: 200G
       redundancyPolicy: "SingleRedundancy"
   visualization:
-    type: "kibana"  <4>
+    type: "kibana"  <5>
     kibana:
       replicas: 1
   curation:
-    type: "curator"  <5>
+    type: "curator"  <6>
     curator:
       schedule: "30 3 * * *"
   collection:
     logs:
-      type: "fluentd"  <6>
+      type: "fluentd"  <7>
       fluentd: {}
 ----
 <1> The name of the CR. This must be `instance`.
 <2> The cluster logging management state. In most cases, if you change the default cluster logging defaults, you must set this to `Unmanaged`.
 However, an unmanaged deployment does not receive updates until the cluster logging is placed back into a managed state. For more information, see *Changing cluster logging management state*.
 <3> Settings for configuring Elasticsearch. Using the CR, you can configure shard replication policy and persistent storage. For more information, see *Configuring Elasticsearch*.
-<4> Settings for configuring Kibana. Using the CR, you can scale Kibana for redundancy and configure the CPU and memory for your Kibana nodes. For more information, see *Configuring Kibana*.
-<5> Settings for configuring Curator. Using the CR, you can set the Curator schedule. For more information, see *Configuring Curator*.
-<6> Settings for configuring Fluentd. Using the CR, you can configure Fluentd CPU and memory limits. For more information, see *Configuring Fluentd*.
+<4> Specify the number of Elasticsearch nodes, maximum of `3`. Any configuration with more than three Elasticsearch nodes is considered invalid and no nodes are created.
```

Comment:
The nodeCount  can >3 now.   The maximum Elasticsearch master nodes(pods) number is 3. That means: when the size >3,  3 elasticsearch nodes(pods) are created as Master-eligible node (using roles: master,client,nodes) , the other ES nodes(pods) will be created as Data only node ( using roles: client,nodes).
Refer to https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html

For example:  when nodeCount=4,
The elasticsearch CRD
  nodes:
  - genUUID: x6kdekli
    nodeCount: 1
    resources: {}
    roles:
    - client
    - data
    storage: {}
  - genUUID: x6kdekli
    nodeCount: 3
    resources: {}
    roles:
    - client
    - data
    - master
    storage: {}
  redundancyPolicy: ZeroRedundancy

The deployment:
[anli@preserve-anli-slave 41]$ oc get deployment
cluster-logging-operator       1/1     1            1           18h
elasticsearch-cd-x6kdekli-1    0/1     1            0           6m54s
elasticsearch-cdm-x6kdekli-1   1/1     1            1           18h
elasticsearch-cdm-x6kdekli-2   0/1     1            0           6m49s
elasticsearch-cdm-x6kdekli-3   0/1     1            0           6m44s

Reply:
@anpingli Thank you. I misread the BZ. How is this:

> The maximum number of Elasticsearch master nodes is three. If you specify a `nodeCount` greater than `3`, {product-title} creates three Elasticsearch nodes that are Master-eligible nodes, with the master, client, and data roles. The additional Elasticsearch nodes are created as Data only node, using client and data roles. Master nodes perform cluster-wide actions such as creating or deleting an index, shard allocation, and tracking nodes. Data nodes hold the shards and perform data-related operations such as CRUD, search, and aggregations. Data-related operations are I/O-, memory-, and CPU-intensive. It is important to monitor these resources and to add more Data nodes if the current nodes are overloaded.

Reply:
LGTM